### PR TITLE
Adds Faros graph source

### PR DESCRIPTION
**Pull request by ypc-faros**
## Description

Adds Faros graph source

TODO: Possibly add page size to spec, include model name in record.

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: sources/faros-graphql-source/package.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    "watch": "tsc -b -w src test"
+  },
+  "dependencies": {
+    "faros-airbyte-cdk": "0.0.1",
+    "faros-airbyte-common": "0.0.1",
+    "faros-js-client": "0.1.4",
+    "verror": "^1.10.1"
+  },
+  "devDependencies": {
+    "@types/jest": "^26.0.14",
```
Comment (cjwooo) [inline sources/faros-graphql-source/package.json:38]: You shouldn't need to specify these devDependencies. Lerna brings in a bunch of `@types/*` and lint/test-related dependencies in the root package.json.

**Review by cjwooo — APPROVED** [review]

File: sources/faros-graphql-source/package.json (hunk 1 above)
Comment (ypc-faros) [reply sources/faros-graphql-source/package.json]: Removed, thanks!!!

File: sources/faros-graphql-source/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+export class FarosGraphSource extends AirbyteSourceBase<GraphQLConfig> {
+  async spec(): Promise<AirbyteSpec> {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    return new AirbyteSpec(require('../resources/spec.json'));
+  }
+
+  async checkConnection(config: GraphQLConfig): Promise<[boolean, VError]> {
+    try {
+      if (!config.api_url) throw new VError('Missing api_url');
```
Comment (tovbinm) [inline sources/faros-graphql-source/src/index.ts:42]: We should make these errors a bit more human readable

File: sources/faros-graphql-source/src/streams/faros-graph.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  toIncrementalV2,
+} from 'faros-js-client';
+import {FarosClient} from 'faros-js-client';
+import {max} from 'lodash';
+import {Dictionary} from 'ts-essentials';
+import VError from 'verror';
+
+import {GraphQLConfig, GraphQLVersion} from '..';
+
+// TODO: Make this part of the spec?
```
Comment (tovbinm) [inline sources/faros-graphql-source/src/streams/faros-graph.ts:20]: Yes please!

File: sources/faros-graphql-source/src/streams/faros-graph.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    streamState?: GraphQLState
+  ): AsyncGenerator<Dictionary<any, string>, any, undefined> {
+    const faros = new FarosClient({
+      url: this.config.api_url,
+      apiKey: this.config.api_key,
+      useGraphQLV2: this.config.graphql_api === GraphQLVersion.V2,
+    });
+    if (!(await faros.graphExists(this.config.graph))) {
+      throw new VError('Graph does not exist!');
+    }
```
Comment (tovbinm) [inline sources/faros-graphql-source/src/streams/faros-graph.ts:65]: This section is repeated here and in check connection. Please DRY

File: sources/faros-graphql-source/src/streams/faros-graph.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+          this.config.graphql_api === GraphQLVersion.V1
+            ? Number(item['metadata']['refreshedAt']) || 0
+            : new Date(item['refreshedAt']).getTime() || 0;
+        refreshedAtMillis = max([refreshedAtMillis, recordRefreshedAtMillis]);
+
+        this.state = {...this.state, [this.config.query]: {refreshedAtMillis}};
+        yield item;
+      }
+    } else {
+      // TODO: Query all models
```
Comment (tovbinm) [inline sources/faros-graphql-source/src/streams/faros-graph.ts:110]: Let's add a log message / or an error that this part is not yet implemented. Or you are going to work on this next?

**Review by tovbinm — COMMENTED** [review]
Lgtm! Very nice and compact!

File: sources/faros-graphql-source/src/streams/faros-graph.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    const faros = new FarosClient({
+      url: this.config.api_url,
+      apiKey: this.config.api_key,
+      useGraphQLV2: this.config.graphql_api === GraphQLVersion.V2,
+    });
+    if (!(await faros.graphExists(this.config.graph))) {
+      throw new VError('Graph does not exist!');
+    }
+
+    if (this.config.query) {
```
Comment (tovbinm) [inline sources/faros-graphql-source/src/streams/faros-graph.ts:67]: It would be great to add some debug logging about: what is the quests being executed.

**Comment by ypc-faros** [comment]
Thanks @tovbinm !!! I'll address this feedback in the following PR

File: sources/faros-graphql-source/src/streams/faros-graph.ts (hunk 5 above)
Comment (ypc-faros) [reply sources/faros-graphql-source/src/streams/faros-graph.ts:110]: Yes, this is next

**Pull request merged**